gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -74,95 +87,128 @@ void parse(String config) throws IOException {
-      Map<String, ?> rawCustomTags = JsonUtil.getObject(config, "custom_tags");
-      if (rawCustomTags != null) {
-        ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
-        for (Map.Entry<String, ?> entry: rawCustomTags.entrySet()) {
-          checkArgument(
-                  entry.getValue() instanceof String,
-                  "'custom_tags' needs to be a map of <string, string>");
-          builder.put(entry.getKey(), (String) entry.getValue());
-        }
-        customTags = builder.build();
+            entry.getValue() instanceof String,
+            "'labels' needs to be a map of <string, string>");
+        builder.put(entry.getKey(), (String) entry.getValue());
       }
+      customTags = builder.build();
     }
   }
 
-  private EventType convertEventType(String val) {
-    switch (val) {
-      case "GRPC_CALL_UNKNOWN":
-        return EventType.GRPC_CALL_UNKNOWN;
-      case "GRPC_CALL_REQUEST_HEADER":
-        return EventType.GRPC_CALL_REQUEST_HEADER;
-      case "GRPC_CALL_RESPONSE_HEADER":
-        return EventType.GRPC_CALL_RESPONSE_HEADER;
-      case "GRPC_CALL_REQUEST_MESSAGE":
-        return EventType.GRPC_CALL_REQUEST_MESSAGE;
-      case "GRPC_CALL_RESPONSE_MESSAGE":
-        return EventType.GRPC_CALL_RESPONSE_MESSAGE;
-      case "GRPC_CALL_TRAILER":
-        return EventType.GRPC_CALL_TRAILER;
-      case "GRPC_CALL_HALF_CLOSE":
-        return EventType.GRPC_CALL_HALF_CLOSE;
-      case "GRPC_CALL_CANCEL":
-        return EventType.GRPC_CALL_CANCEL;
-      default:
-        throw new IllegalArgumentException("Unknown event type value:" + val);
+  private List<LogFilter> parseRpcEvents(List<?> rpcEvents) {
+    List<Map<String, ?>> jsonRpcEvents = JsonUtil.checkObjectList(rpcEvents);
+    ImmutableList.Builder<LogFilter> rpcEventsListBuilder =
+        new ImmutableList.Builder<>();
+    for (Map<String, ?> jsonClientRpcEvent : jsonRpcEvents) {
+      rpcEventsListBuilder.add(parseJsonLogFilter(jsonClientRpcEvent));
     }
+    return rpcEventsListBuilder.build();
   }
 
+
   private LogFilter parseJsonLogFilter(Map<String, ?> logFilterMap) {
-    return new LogFilter(JsonUtil.getString(logFilterMap, "pattern"),
-        JsonUtil.getNumberAsInteger(logFilterMap, "header_bytes"),
-        JsonUtil.getNumberAsInteger(logFilterMap, "message_bytes"));
+    Boolean exclude = JsonUtil.getBoolean(logFilterMap, "exclude");
+    Boolean global = false;
+    ImmutableList.Builder<String> patternListBuilder =
+        new ImmutableList.Builder<>();
+    ImmutableSet.Builder<String> servicesSetBuilder =
+        new ImmutableSet.Builder<>();
+    ImmutableSet.Builder<String> methodsSetBuilder =
+        new ImmutableSet.Builder<>();
+    List<?> methodsList = JsonUtil.getList(logFilterMap, "methods");
+    if (methodsList != null) {
+      for (Object pattern : methodsList) {
+        String methodOrServicePattern = String.valueOf(pattern);
+        if (methodOrServicePattern != null) {
```

Comment:
This is always true. And will still always be true if swapped to getListOfStrings() above.

Reply:
Updated